src/Umbraco.Web/Editors/IconController.cs

```diff
@@ -34,5 +41,18 @@ public IList<IconModel> GetAllIcons()
         {
             return _iconService.GetAllIcons();
         }
+
+        /// <summary>
+        /// Gets JSON blob containing all the known icons
+        /// </summary>
+        /// <param name="culture"></param>
+        /// <returns></returns>
+        public JsonNetResult GetIcons()
+        {
+            var dictionary = _iconService.GetAllIcons()
```

Comment:
Extending Iconservice to return this dictionary would mean not having to build the dictionary on every request.

Reply:
Indeed it would - there's a cached `Dictionary<string, IconModel>` available in the service, ideally we'd update that to be `<string, string>` but would need to check where the existing dictionary is used...

Reply:
Perhaps use IReadonlyDictionary